extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/proxies/ProxyDefinitions.java

```diff
@@ -49,7 +49,7 @@ private ProxyDefinitions(Map<Class<?>, ProxyClassDetailsHolder> proxyDefinitionM
         this.proxyDefinitionMap = proxyDefinitionMap;
     }
 
-    public static ProxyDefinitions createFromMetadata(Metadata storeableMetadata) {
+    public static ProxyDefinitions createFromMetadata(Metadata storeableMetadata, PreGeneratedProxies preGeneratedProxies) {
         //Check upfront for any need across all metadata: would be nice to avoid initializing the Bytecode provider.
```

Comment:
There's an important optimisation missing here: even if all proxies are generated upfront, we still initialize the Bytecode Provider!

The line `final BytecodeProviderImpl bytecodeProvider = new BytecodeProviderImpl();` needs to become lazily evaulates if we aim to skip loading all those classes.